api/data_test.go

```diff
@@ -149,3 +149,88 @@ func TestCreateNewCacheTime(t *testing.T) {
 		})
 	})
 }
+
+func TestCreateOrUpdateCacheTimeReturnsErr(t *testing.T) {
+	const validBody = `{"path": "testpath", "etag": "testetag"}`
+	Convey("When no request body is provided, 400 bad request is returned", t, func() {
+		ctx := context.Background()
+		dataStoreMock := &mock.DataStoreMock{}
+		body := ""
+		request := httptest.NewRequest(http.MethodPut, baseURL+testCacheID, bytes.NewBufferString(body))
```

Comment:
I think you can use `http.NoBody` here instead (as the third parameter of `NewRequest`).

Reply:
Interestingly, using `http.NoBody` actually sets the `req.ContentLength` to `-1`. So I've updated the code to catch both scenarios.